kubejs/server_scripts/enigmatica/kubejs/base/tags/items/enigmatica/explosives.js

```diff
@@ -0,0 +1,99 @@
+     * @type {{tag:string,firstRemove?:any|any[],thenAdd?:any|any[]}[]}
+     * @param tag the tag string WITHOUT `#` prefix, like `forge:ingots` or `why:using/this/tag`
+     * @param firstRemove (Optional) Accepts RegEx, tag string, item string.
+     * @param thenAdd (Optional) Accepts can use RegEx, tag string, item string.
+     */
+    let recipes = [
+        {
+            tag: 'enigmatica:explosives/base',
+            thenAdd: [
+                'minecraft:tnt',
+                'minecraft:tnt_minecart',
+                'thermal:explosive_grenade',
+                'supplementaries:bomb',
+                'archers_paradox:explosive_arrow',
+                'immersiveengineering:he',
+                'apotheosis:explosive_arrow',
+                'botania:lens_explosive'
+            ]
+        },
+        {
+            tag: 'enigmatica:explosives/lightning',
+            thenAdd: [
+                'powah:charged_snowball',
+                'thermal:lightning_charge',
+                'thermal:lightning_grenade',
+                'thermal:lightning_tnt',
+                'thermal:lightning_tnt_minecart',
+                'archers_paradox:lightning_arrow'
+            ]
+        },
+        {
+            tag: 'enigmatica:explosives/ice',
+            thenAdd: [
+                'thermal:ice_charge',
+                'thermal:ice_grenade',
+                'thermal:ice_tnt',
+                'thermal:ice_tnt_minecart'
+            ]
+        },
+        {
+            tag: 'enigmatica:explosives/earth',
+            thenAdd: [
+                'tconstruct:efln_ball',
+                'thermal:earth_charge',
+                'thermal:earth_tnt',
+                'thermal:earth_tnt_minecart',
+                'thermal:earth_grenade'
+            ]
+        },
+        {
+            tag: 'enigmatica:explosives/slime',
+            thenAdd: ['thermal:slime_tnt_minecart', 'thermal:slime_tnt', 'thermal:slime_grenade']
+        },
+        {
+            tag: 'enigmatica:explosives/fire',
+            thenAdd: [
+                'thermal:fire_grenade',
+                'minecraft:fire_charge',
+                'thermal:fire_tnt',
+                'thermal:fire_tnt_minecart',
+                'archers_paradox:blaze_arrow'
+            ]
+        },
+        {
+            tag: 'enigmatica:explosives/ender',
+            thenAdd: ['thermal:ender_grenade', 'thermal:ender_tnt', 'thermal:ender_tnt_minecart']
+        },
+        {
+            tag: 'enigmatica:explosives/glow',
+            thenAdd: ['thermal:glowstone_grenade', 'thermal:glowstone_tnt', 'thermal:glowstone_tnt_minecart']
+        },
+        {
+            tag: 'enigmatica:explosives/redstone',
+            thenAdd: [
+                'thermal:redstone_grenade',
+                'thermal:redstone_tnt',
+                'thermal:redstone_tnt_minecart',
+                'archers_paradox:redstone_arrow'
+            ]
+        }
+    ];
+
+    for (let recipe of recipes) {
```

Comment:
We normally use the for each syntax `recipes.ForEach(recipe => {...})`
I'd like this to be the same, for the sake of consistency.